src/cmd/services/m3coordinator/downsample/metrics_appender.go

```diff
@@ -109,6 +117,12 @@ func (a *metricsAppender) SamplesAppender(opts SampleAppenderOptions) (SamplesAp
 	matchResult := a.matcher.ForwardMatch(id, fromNanos, toNanos)
 	id.Close()
 
+	// Filter out the m3 custom tags once the ForwardMatch is complete. If
+	// modified set the unowned ID to nil so that we know to recreate it later.
```

Comment:
nit: Can we use a bool here instead? Using nil as implicit state seems a little less readable.
i.e. 
```go
needRecomputeID := false

//... 

if modified := a.tags.filterPrefix(m3MetricsPrefix); modified {
  needRecomputeID = true
}

if needRecomputeID {
  unownedID = ....
}
```

Reply:
let me clean this up.. That's actually not used anywhere and we just compute it before we send stuff to aggregator. Also, the filter will always filter out stuff as we add the metrics type tag all the time..